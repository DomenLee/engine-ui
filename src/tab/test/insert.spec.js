import { mount, later } from '../../../test';

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab dynamically', async () => {
  const wrapper = mount({
    template: `
      <ghb-tabs v-model="active">
        <ghb-tab title="1">1</ghb-tab>
        <div v-if="insert">
          <ghb-tab title="2">2</ghb-tab>
        </div>
        <ghb-tab title="3">3</ghb-tab>
      </ghb-tabs>
    `,
    data() {
      return {
        insert: false,
        active: 1,
      };
    },
  });

  await later();
  wrapper.setData({ insert: true });
  expect(wrapper).toMatchSnapshot();
});

test('insert tab with name dynamically', async () => {
  const onChange = jest.fn();
  const wrapper = mount({
    template: `
      <ghb-tabs v-model="active" @change="onChange">
        <ghb-tab v-if="insert" title="1" name="bar">2</ghb-tab>
        <ghb-tab title="2" name="foo">1</ghb-tab>
      </ghb-tabs>
    `,
    data() {
      return {
        insert: false,
        active: 'foo',
      };
    },
    methods: {
      onChange,
    },
  });

  await later();
  wrapper.setData({ insert: true });
  expect(wrapper).toMatchSnapshot();
  expect(onChange).toHaveBeenCalledTimes(0);
});

// this case will throw wierd error in index.spec.js
// so separate it
test('insert tab with child component', async () => {
  const wrapper = mount({
    template: `
      <ghb-tabs v-model="active">
        <ghb-tab title="1">1</ghb-tab>
        <my-tab />
        <ghb-tab title="3">3</ghb-tab>
      </ghb-tabs>
    `,
    components: {
      'my-tab': {
        template: `<ghb-tab title="2">2</ghb-tab>`,
      },
    },
  });

  await later();
  expect(wrapper).toMatchSnapshot();
});
