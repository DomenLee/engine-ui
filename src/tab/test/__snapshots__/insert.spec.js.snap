// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert tab dynamically 1`] = `
<div class="ghb-tabs ghb-tabs--line">
  <div class="ghb-tabs__wrap">
    <div role="tablist" class="ghb-tabs__nav ghb-tabs__nav--line">
      <div role="tab" class="ghb-tab"><span class="ghb-tab__text ghb-tab__text--ellipsis">1</span></div>
      <div role="tab" aria-selected="true" class="ghb-tab ghb-tab--active"><span class="ghb-tab__text ghb-tab__text--ellipsis">2</span></div>
      <div role="tab" class="ghb-tab"><span class="ghb-tab__text ghb-tab__text--ellipsis">3</span></div>
      <div class="ghb-tabs__line" style="transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="ghb-tabs__content">
    <div role="tabpanel" class="ghb-tab__pane" style="display: none;">
      <!---->
    </div>
    <div>
      <div role="tabpanel" class="ghb-tab__pane">2</div>
    </div>
    <div role="tabpanel" class="ghb-tab__pane" style="display: none;">3</div>
  </div>
</div>
`;

exports[`insert tab with child component 1`] = `
<div class="ghb-tabs ghb-tabs--line">
  <div class="ghb-tabs__wrap">
    <div role="tablist" class="ghb-tabs__nav ghb-tabs__nav--line">
      <div role="tab" aria-selected="true" class="ghb-tab ghb-tab--active"><span class="ghb-tab__text ghb-tab__text--ellipsis">1</span></div>
      <div role="tab" class="ghb-tab"><span class="ghb-tab__text ghb-tab__text--ellipsis">2</span></div>
      <div role="tab" class="ghb-tab"><span class="ghb-tab__text ghb-tab__text--ellipsis">3</span></div>
      <div class="ghb-tabs__line"></div>
    </div>
  </div>
  <div class="ghb-tabs__content">
    <div role="tabpanel" class="ghb-tab__pane">1</div>
    <div role="tabpanel" class="ghb-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="ghb-tab__pane" style="display: none;">
      <!---->
    </div>
  </div>
</div>
`;

exports[`insert tab with name dynamically 1`] = `
<div class="ghb-tabs ghb-tabs--line">
  <div class="ghb-tabs__wrap">
    <div role="tablist" class="ghb-tabs__nav ghb-tabs__nav--line">
      <div role="tab" class="ghb-tab"><span class="ghb-tab__text ghb-tab__text--ellipsis">1</span></div>
      <div role="tab" aria-selected="true" class="ghb-tab ghb-tab--active"><span class="ghb-tab__text ghb-tab__text--ellipsis">2</span></div>
      <div class="ghb-tabs__line" style="transform: translateX(0px) translateX(-50%);"></div>
    </div>
  </div>
  <div class="ghb-tabs__content">
    <div role="tabpanel" class="ghb-tab__pane" style="display: none;">
      <!---->
    </div>
    <div role="tabpanel" class="ghb-tab__pane" style="">1</div>
  </div>
</div>
`;
